python/ray/air/checkpoint.py

```diff
@@ -351,6 +351,15 @@ def from_directory(cls, path: str) -> "Checkpoint":
         """
         return cls(local_path=path)
 
+    @classmethod
+    def _from_checkpoint(cls, other: "Checkpoint") -> "Checkpoint":
```

Comment:
```suggestion
    def copy_from(cls, other: "Checkpoint") -> "Checkpoint":
```

Reply:
Could we make the method private? I'm not sure whether we should expose this functionality to the user.

```suggestion
    def _copy_from(cls, other: "Checkpoint") -> "Checkpoint":
```

Reply:
We need to expose the cast operation to the user. So either we expose `copy_from()`, or we expose it in the constructor of the framework-specific checkpoint (we should pick one or the other).